### PR TITLE
Update add GroupUser permissions checking

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: skyportal/handlers/api/group.py [hunk 1]
```diff
@@ -354,6 +354,22 @@ def post(self, group_id, *ignored_args):
                               type: boolean
                               description: Boolean indicating whether user is group admin
         """
+        current_groupuser = (
+            GroupUser.query.filter(GroupUser.group_id == group_id)
+            .filter(GroupUser.user_id == self.current_user.id)
```
Comment (dannygoldstein) [inline skyportal/handlers/api/group.py:359]:
```suggestion
            .filter(GroupUser.user_id == self.associated_user_object.id)
```

File: skyportal/handlers/api/group.py [hunk 2]
```diff
@@ -354,6 +354,22 @@ def post(self, group_id, *ignored_args):
...
+            .first()
+        )
+        if (
+            len(
+                {"System admin", "Manage users", "Manage groups"}.intersection(
+                    set(self.current_user.permissions)
+                )
+            )
+            == 0
+            and not current_groupuser.admin
```
Comment (dannygoldstein) [inline skyportal/handlers/api/group.py:369]: If a user is an admin, would they not also have some or all of the 3 ACLs above?

File: skyportal/handlers/api/group.py [hunk 3]
```diff
@@ -438,6 +454,22 @@ def delete(self, group_id, username):
...
                 schema: Success
         """
+        current_groupuser = (
+            GroupUser.query.filter(GroupUser.group_id == group_id)
+            .filter(GroupUser.user_id == self.current_user.id)
+            .first()
+        )
+        if (
+            len(
+                {"System admin", "Manage users", "Manage groups"}.intersection(
```
Comment (dannygoldstein) [inline skyportal/handlers/api/group.py:464]: See comment above

File: skyportal/handlers/api/group.py [hunk 4]
```diff
@@ -438,6 +454,22 @@ def delete(self, group_id, username):
...
+                    set(self.current_user.permissions)
+                )
+            )
+            == 0
+            and not current_groupuser.admin
```
Comment (dannygoldstein) [inline skyportal/handlers/api/group.py:469]: See comment above

**Review by dannygoldstein — CHANGES_REQUESTED** [review]

File: skyportal/handlers/api/group.py (hunk 2 above)
Comment (acrellin) [reply skyportal/handlers/api/group.py:369]: No they would not necessarily -- a "Full user" will have none of these ACLs, but can still be an admin for a specific group

File: skyportal/handlers/api/group.py (hunk 1 above)
Comment (acrellin) [reply skyportal/handlers/api/group.py]: Done

**Comment by acrellin** [comment]
@dannygoldstein thanks for the review. I incorporated some of your suggested changes -- others I responded to inline: 
No they would not necessarily -- a "Full user" will have none of these ACLs, but can still be an admin for a specific group

**Review by dannygoldstein — APPROVED** [review]

**Comment by dannygoldstein** [comment]
@acrellin looks like some real failures related to uploading photometry

**Comment by acrellin** [comment]
@dannygoldstein Those are definitely not related to the changes here.
Re-running.

**Comment by acrellin** [comment]
@dannygoldstein the only failures here are also present on master. Check out this TB for the upload photometry failure (here and on master):
```
ERROR:tornado.application:Uncaught exception POST /api/photometry (127.0.0.1)                                                              
HTTPServerRequest(protocol='http', host='localhost:5000', method='POST', uri='/api/photometry', version='HTTP/1.0', remote_ip='127.0.0.1') 
Traceback (most recent call last):                                                                                                         
  File "/home/arien/anaconda3/envs/skyportal38/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute                         
    result = method(*self.path_args, **self.path_kwargs)                                                                                   
  File "/home/arien/projects/skyportal/baselayer/app/access.py", line 32, in wrapper                                                       
    return tornado.web.authenticated(method)(self, *args, **kwargs)                                                                        
  File "/home/arien/anaconda3/envs/skyportal38/lib/python3.8/site-packages/tornado/web.py", line 3174, in wrapper                          
    return method(self, *args, **kwargs)                                                                                                   
  File "/home/arien/projects/skyportal/baselayer/app/access.py", line 51, in wrapper                                                       
    return method(self, *args, **kwargs)                                                                                                   
  File "/home/arien/projects/skyportal/skyportal/handlers/api/photometry.py", line 365, in post                                            
    raise ValidationError(                                                                                                                 
marshmallow.exceptions.ValidationError: Instrument ZTF_3c016a2d-8a24-4025-b35a-61e0ebfd8d40 has no filter sdssg.                           
[!] Error in `/api/photometry`: Instrument ZTF_3c016a2d-8a24-4025-b35a-61e0ebfd8d40 has no filter sdssg.                                   
WARNING:tornado.access:400 POST /api/photometry (127.0.0.1) 33.65ms                                                                        
```
Merging, as you've approved and this does not introduce test failures.

**Pull request merged**